test/e2e/specs/specs-playwright/wp-theme__details-preview-activate.js

```diff
@@ -33,6 +34,16 @@ describe( DataHelper.createSuiteTitle( 'Theme: Preview and Activate' ), () => {
 		await sidebarComponent.gotoMenu( { item: 'Appearance', subitem: 'Themes' } );
 	} );
 
+	it( 'Select test site', async function () {
+		const siteSelectComponent = new SiteSelectComponent( page );
+
+		if ( await siteSelectComponent.siteSelectorShown() ) {
+			await siteSelectComponent.selectSite(
+				DataHelper.getAccountSiteURL( user, { protocol: false } )
+			);
+		}
+	} );
```

Comment:
Just to double check: is where you were seeing this issue in previous runs is when you tried to navigate to themes? So this should come _after_ the sidebar navigation?

Reply:
You are correct - the select site component appears after navigating to a particular feature that requires a site to be selected. In this case, the check takes place after the sidebar navigation to Appearance > Themes, to see if the site selector is loaded prior to executing rest of the theme test case.